k8s/helm-chart/dynatrace-gcp-function/templates/active-gate-pod.yaml

```diff
@@ -0,0 +1,108 @@
+# Copyright 2021 Dynatrace LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+{{- if or (eq .Values.deploymentType "logs") (eq .Values.deploymentType "all") }}
+  {{- if (.Values.activeGate.autoDeploy) }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: dynatrace-activegate-gcpmon-router
+  namespace: dynatrace
+spec:
+  type: ClusterIP
+  selector:
+    app: dynatrace-ag-gcpmon
+  ports:
+    - protocol: TCP
+      port: {{ .Values.activeGate.autoDeployPort }}
+      targetPort: ag-https
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: dynatrace-gcpmon-pullsecret
+  namespace: dynatrace
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: {{ template "imagePullSecret" . }}
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: dynatrace-ag-gcpmon
+  labels:
+    app: dynatrace-ag-gcpmon
+  namespace: dynatrace
+spec:
+  selector:
+    matchLabels:
+      app: dynatrace-ag-gcpmon
+  replicas: 1
+  serviceName: dynatrace-ag-gcpmon
```

Comment:
Did you come up with any solution for long startup time and synchronization with Dyantrace - GCP Function pods?

Reply:
Working on it. For now, I'll try an ``` initConteiner ``` and check for RUNNING state of  AG pod.

Reply:
The init Container worked as expected.